File: samples-classic/javascript/cleanup.js
#!/usr/bin/env node
/**
 * Centralized cleanup utility for samples-classic TypeScript/JavaScript samples
 * 
 * This script provides utilities to clean up agents, threads, vector stores, and files
 * created during sample demos in the samples-classic directory.
 * 
 * Usage:
 *   node cleanup.js --agent <agent_id>
 *   node cleanup.js --thread <thread_id>
 *   node cleanup.js --vector-store <vector_store_id>
 *   node cleanup.js --file <file_id>
 *   node cleanup.js --agent <id> --thread <id> --file <id>
 *   node cleanup.js --list-agents
 *   node cleanup.js --clean-all
 * 
 * Environment Variables:
 *   PROJECT_ENDPOINT or AZURE_AI_PROJECT_ENDPOINT: Your AI Project endpoint
 *   PROJECT_CONNECTION_STRING: Your AI Project connection string (alternative)
 */

const { AIProjectClient } = require("@azure/ai-projects");
const { DefaultAzureCredential } = require("@azure/identity");

function getProjectClient() {
  const endpoint = 
    process.env.PROJECT_ENDPOINT ||
    process.env.AZURE_AI_PROJECT_ENDPOINT ||
    process.env.AIPROJECT_CONNECTION_STRING;

  if (!endpoint) {
    console.error("❌ Error: No project endpoint found in environment variables.");
    console.error("Set one of: PROJECT_ENDPOINT, AZURE_AI_PROJECT_ENDPOINT, or AIPROJECT_CONNECTION_STRING");
    process.exit(1);
  }

  try {
    const credential = new DefaultAzureCredential();
    return AIProjectClient.fromConnectionString(endpoint, credential);
  } catch (error) {
    console.error(`❌ Error creating project client: ${error.message}`);
    process.exit(1);
  }
}

async function deleteAgent(client, agentId) {
  try {
    console.log(`🗑️  Deleting agent '${agentId}'...`);
    await client.agents.deleteAgent(agentId);
    console.log(`✅ Successfully deleted agent '${agentId}'`);
    return true;
  } catch (error) {
    if (error.statusCode === 404 || error.status === 404) {
      console.log(`⚠️  Agent '${agentId}' not found (may have been already deleted)`);
    } else {
      console.error(`❌ Error deleting agent '${agentId}': ${error.message}`);
    }
    return false;
  }
}

async function deleteThread(client, threadId) {
  try {
    console.log(`🗑️  Deleting thread '${threadId}'...`);
    await client.agents.threads.delete(threadId);
    console.log(`✅ Successfully deleted thread '${threadId}'`);
    return true;
  } catch (error) {
    if (error.statusCode === 404 || error.status === 404) {
      console.log(`⚠️  Thread '${threadId}' not found (may have been already deleted)`);
    } else {
      console.error(`❌ Error deleting thread '${threadId}': ${error.message}`);
    }
    return false;
  }
}

async function deleteVectorStore(client, vectorStoreId) {
  try {
    console.log(`🗑️  Deleting vector store '${vectorStoreId}'...`);
    await client.agents.vectorStores.delete(vectorStoreId);
    console.log(`✅ Successfully deleted vector store '${vectorStoreId}'`);
    return true;
  } catch (error) {
    if (error.statusCode === 404 || error.status === 404) {
      console.log(`⚠️  Vector store '${vectorStoreId}' not found (may have been already deleted)`);
    } else {
      console.error(`❌ Error deleting vector store '${vectorStoreId}': ${error.message}`);
    }
    return false;
  }
}

async function deleteFile(client, fileId) {
  try {
    console.log(`🗑️  Deleting file '${fileId}'...`);
    await client.agents.files.delete(fileId);
    console.log(`✅ Successfully deleted file '${fileId}'`);
    return true;
  } catch (error) {
    if (error.statusCode === 404 || error.status === 404) {
      console.log(`⚠️  File '${fileId}' not found (may have been already deleted)`);
    } else {
      console.error(`❌ Error deleting file '${fileId}': ${error.message}`);
    }
    return false;
  }
}

async function listAgents(client) {
  try {
    console.log("📋 Listing all agents...");
    const agents = [];
    for await (const agent of client.agents.list()) {
      agents.push(agent);
    }
    
    if (agents.length > 0) {
      console.log(`\nFound ${agents.length} agent(s):`);
      for (const agent of agents) {
        console.log(`  - ID: ${agent.id}, Name: ${agent.name || '(unnamed)'}, Created: ${agent.createdAt}`);
      }
    } else {
      console.log("ℹ️  No agents found");
    }
    return agents;
  } catch (error) {
    console.error(`❌ Error listing agents: ${error.message}`);
    return [];
  }
}

async function cleanAll(client) {
  console.log("⚠️  WARNING: This will attempt to delete ALL agents in your project.");
  
  // Simple synchronous prompt for Node.js
  const readline = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout
  });

  return new Promise((resolve) => {
    readline.question("Are you sure you want to continue? (yes/no): ", async (answer) => {
      readline.close();
      
      if (answer.toLowerCase() !== "yes") {
        console.log("❌ Cancelled");
        resolve();
        return;
      }

      const agents = await listAgents(client);
      if (agents.length === 0) {
        resolve();
        return;
      }

      let deletedCount = 0;
      for (const agent of agents) {
        if (await deleteAgent(client, agent.id)) {
          deletedCount++;
        }
      }

      console.log(`\n✅ Cleanup complete: ${deletedCount} agent(s) deleted`);
      resolve();
    });
  });
}

async function main() {
  const args = process.argv.slice(2);
  
  const options = {
    agent: null,
    thread: null,
    vectorStore: null,
    file: null,
    listAgents: false,
    cleanAll: false
  };

  // Parse command line arguments
  for (let i = 0; i < args.length; i++) {
    const arg = args[i];
    
    if (arg === "--agent" && i + 1 < args.length) {
      options.agent = args[++i];
    } else if (arg === "--thread" && i + 1 < args.length) {
      options.thread = args[++i];
    } else if (arg === "--vector-store" && i + 1 < args.length) {
      options.vectorStore = args[++i];
    } else if (arg === "--file" && i + 1 < args.length) {
      options.file = args[++i];
    } else if (arg === "--list-agents") {
      options.listAgents = true;
    } else if (arg === "--clean-all") {
      options.cleanAll = true;
    } else if (arg === "--help" || arg === "-h") {
      console.log(`
Samples-Classic TypeScript/JavaScript - Resource Cleanup Utility

Usage:
  node cleanup.js --agent <agent_id>
  node cleanup.js --thread <thread_id>
  node cleanup.js --vector-store <vector_store_id>
  node cleanup.js --file <file_id>
  node cleanup.js --agent <id> --thread <id> --file <id>
  node cleanup.js --list-agents
  node cleanup.js --clean-all

Options:
  --agent <id>          Delete an agent by ID
  --thread <id>         Delete a thread by ID
  --vector-store <id>   Delete a vector store by ID
  --file <id>          Delete a file by ID
  --list-agents        List all agents
  --clean-all          Delete all agents (interactive confirmation)
  --help, -h           Show this help message

Environment Variables:
  PROJECT_ENDPOINT or AZURE_AI_PROJECT_ENDPOINT: Your AI Project endpoint
      `);
      process.exit(0);
    }
  }

  // Check if any action was specified
  const hasAction = options.agent || options.thread || options.vectorStore || 
                    options.file || options.listAgents || options.cleanAll;
  
  if (!hasAction) {
    console.log("Use --help for usage information");
    process.exit(0);
  }

  console.log("🧹 Samples-Classic TypeScript/JavaScript - Resource Cleanup Utility");
  console.log("=".repeat(60));

  const client = getProjectClient();

  try {
    let cleanupPerformed = false;

    if (options.listAgents) {
      await listAgents(client);
      cleanupPerformed = true;
    }

    if (options.cleanAll) {
      await cleanAll(client);
      cleanupPerformed = true;
    }

    if (options.agent) {
      await deleteAgent(client, options.agent);
      cleanupPerformed = true;
    }

    if (options.thread) {
      await deleteThread(client, options.thread);
      cleanupPerformed = true;
    }

    if (options.vectorStore) {
      await deleteVectorStore(client, options.vectorStore);
      cleanupPerformed = true;
    }

    if (options.file) {
      await deleteFile(client, options.file);
      cleanupPerformed = true;
    }

    if (cleanupPerformed) {
      console.log("=".repeat(60));
      console.log("✅ Cleanup utility finished");
    }
  } catch (error) {
    console.error(`\n❌ Cleanup failed: ${error.message}`);
    process.exit(1);
  }
}

main();

File: samples-classic/python/cleanup.py
"""
Centralized cleanup utility for samples-classic Python samples.

This script provides utilities to clean up agents, threads, vector stores, and files
created during sample demos in the samples-classic directory.

Usage:
    # Clean up a specific agent
    python cleanup.py --agent <agent_id>
    
    # Clean up a thread
    python cleanup.py --thread <thread_id>
    
    # Clean up a vector store
    python cleanup.py --vector-store <vector_store_id>
    
    # Clean up a file
    python cleanup.py --file <file_id>
    
    # Clean up multiple resources
    python cleanup.py --agent <id> --thread <id> --file <id>
    
    # List all agents
    python cleanup.py --list-agents
    
    # Clean up all resources (interactive confirmation)
    python cleanup.py --clean-all

Environment Variables:
    PROJECT_ENDPOINT or AZURE_AI_PROJECT_ENDPOINT: Your AI Project endpoint
    PROJECT_CONNECTION_STRING: Your AI Project connection string (alternative)
"""

import os
import sys
import argparse
from typing import Optional, List

try:
    from azure.ai.projects import AIProjectClient
    from azure.identity import DefaultAzureCredential
    from azure.core.exceptions import ResourceNotFoundError, HttpResponseError
except ImportError:
    print("Error: Required packages not installed.")
    print("Install with: pip install azure-ai-projects azure-identity")
    sys.exit(1)


def get_project_client() -> AIProjectClient:
    """Create and return an AI Project client."""
    # Try different environment variable names
    endpoint = (
        os.getenv("PROJECT_ENDPOINT") or 
        os.getenv("AZURE_AI_PROJECT_ENDPOINT") or
        os.getenv("AIPROJECT_CONNECTION_STRING")
    )
    
    if not endpoint:
        print("❌ Error: No project endpoint found in environment variables.")
        print("Set one of: PROJECT_ENDPOINT, AZURE_AI_PROJECT_ENDPOINT, or AIPROJECT_CONNECTION_STRING")
        sys.exit(1)
    
    try:
        credential = DefaultAzureCredential()
        return AIProjectClient(endpoint=endpoint, credential=credential)
    except Exception as e:
        print(f"❌ Error creating project client: {e}")
        sys.exit(1)


def delete_agent(client: AIProjectClient, agent_id: str) -> bool:
    """Delete an agent by ID."""
    try:
        print(f"🗑️  Deleting agent '{agent_id}'...")
        client.agents.delete_agent(agent_id)
        print(f"✅ Successfully deleted agent '{agent_id}'")
        return True
    except ResourceNotFoundError:
        print(f"⚠️  Agent '{agent_id}' not found (may have been already deleted)")
        return False
    except Exception as e:
        print(f"❌ Error deleting agent '{agent_id}': {e}")
        return False


def delete_thread(client: AIProjectClient, thread_id: str) -> bool:
    """Delete a thread by ID."""
    try:
        print(f"🗑️  Deleting thread '{thread_id}'...")
        client.agents.threads.delete(thread_id)
        print(f"✅ Successfully deleted thread '{thread_id}'")
        return True
    except ResourceNotFoundError:
        print(f"⚠️  Thread '{thread_id}' not found (may have been already deleted)")
        return False
    except Exception as e:
        print(f"❌ Error deleting thread '{thread_id}': {e}")
        return False


def delete_vector_store(client: AIProjectClient, vector_store_id: str) -> bool:
    """Delete a vector store by ID."""
    try:
        print(f"🗑️  Deleting vector store '{vector_store_id}'...")
        client.agents.vector_stores.delete(vector_store_id)
        print(f"✅ Successfully deleted vector store '{vector_store_id}'")
        return True
    except ResourceNotFoundError:
        print(f"⚠️  Vector store '{vector_store_id}' not found (may have been already deleted)")
        return False
    except Exception as e:
        print(f"❌ Error deleting vector store '{vector_store_id}': {e}")
        return False


def delete_file(client: AIProjectClient, file_id: str) -> bool:
    """Delete a file by ID."""
    try:
        print(f"🗑️  Deleting file '{file_id}'...")
        client.agents.files.delete(file_id)
        print(f"✅ Successfully deleted file '{file_id}'")
        return True
    except ResourceNotFoundError:
        print(f"⚠️  File '{file_id}' not found (may have been already deleted)")
        return False
    except Exception as e:
        print(f"❌ Error deleting file '{file_id}': {e}")
        return False


def list_agents(client: AIProjectClient) -> List:
    """List all agents."""
    try:
        print("📋 Listing all agents...")
        agents = list(client.agents.list())
        if agents:
            print(f"\nFound {len(agents)} agent(s):")
            for agent in agents:
                print(f"  - ID: {agent.id}, Name: {agent.name or '(unnamed)'}, Created: {agent.created_at}")
        else:
            print("ℹ️  No agents found")
        return agents
    except Exception as e:
        print(f"❌ Error listing agents: {e}")
        return []


def clean_all(client: AIProjectClient):
    """Interactively clean up all resources."""
    print("⚠️  WARNING: This will attempt to delete ALL agents in your project.")
    response = input("Are you sure you want to continue? (yes/no): ")
    
    if response.lower() != "yes":
        print("❌ Cancelled")
        return
    
    agents = list_agents(client)
    if not agents:
        return
    
    deleted_count = 0
    for agent in agents:
        if delete_agent(client, agent.id):
            deleted_count += 1
    
    print(f"\n✅ Cleanup complete: {deleted_count} agent(s) deleted")


def main():
    parser = argparse.ArgumentParser(
        description="Clean up resources created by samples-classic Python samples",
        formatter_class=argparse.RawDescriptionHelpFormatter,
        epilog="""
Examples:
  python cleanup.py --agent asst_abc123
  python cleanup.py --agent asst_abc123 --thread thread_xyz789
  python cleanup.py --list-agents
  python cleanup.py --clean-all
        """
    )
    
    parser.add_argument("--agent", help="Agent ID to delete")
    parser.add_argument("--thread", help="Thread ID to delete")
    parser.add_argument("--vector-store", help="Vector store ID to delete")
    parser.add_argument("--file", help="File ID to delete")
    parser.add_argument("--list-agents", action="store_true", help="List all agents")
    parser.add_argument("--clean-all", action="store_true", help="Delete all agents (interactive)")
    
    args = parser.parse_args()
    
    # Check if any action was specified
    if not any([args.agent, args.thread, args.vector_store, args.file, args.list_agents, args.clean_all]):
        parser.print_help()
        sys.exit(0)
    
    print("🧹 Samples-Classic Python - Resource Cleanup Utility")
    print("=" * 60)
    
    client = get_project_client()
    
    cleanup_performed = False
    
    with client:
        if args.list_agents:
            list_agents(client)
            cleanup_performed = True
        
        if args.clean_all:
            clean_all(client)
            cleanup_performed = True
        
        if args.agent:
            delete_agent(client, args.agent)
            cleanup_performed = True
        
        if args.thread:
            delete_thread(client, args.thread)
            cleanup_performed = True
        
        if args.vector_store:
            delete_vector_store(client, args.vector_store)
            cleanup_performed = True
        
        if args.file:
            delete_file(client, args.file)
            cleanup_performed = True
    
    if cleanup_performed:
        print("=" * 60)
        print("✅ Cleanup utility finished")


if __name__ == "__main__":
    main()
